### DISCUSSION TITLE
Is there a way to search all valid provider names?

### DISCUSSION THREAD

**Discussion by asteiker** [earthaccess-dev/earthaccess#615] [Q&A]
I'm following the documentation here for opening a url or list of urls: https://earthaccess.readthedocs.io/en/latest/user-reference/api/api/#earthaccess.api.open. It says that I need to specify a provider. Is there a way to search available providers using earthaccess? Or better yet, a way to find the provider based on the URL that I have?

**Comment by mfisher87** [comment]
We say it's optional in the docs, but we don't say what the behavior is if you don't provide it. :thinking:
